src/include/cpu/rpp_cpu_simd.hpp

```diff
@@ -1021,6 +1030,111 @@ inline void rpp_load48_u8pkd3_to_f32pln3_avx(Rpp8u *srcPtr, __m256 *p)
+}
+
+inline void rpp_glitch_load24_f32pkd3_to_f32pln3_avx(Rpp32f *srcPtr, __m256 *p, int *srcLocs)
+{
+    __m128 p128[8];
+    for(int i = 0; i < 3; i++)
+    {
+        p128[0] = _mm_loadu_ps(srcPtr + srcLocs[i]);            /* load [R01|G01|B01|R02] - Need R01 */
+        p128[1] = _mm_loadu_ps(srcPtr + srcLocs[i] + 3);        /* load [R02|G02|B02|R03] - Need R01 */
+        p128[2] = _mm_loadu_ps(srcPtr + srcLocs[i] + 6);        /* load [R03|G03|B03|R04] - Need R03 */
+        p128[3] = _mm_loadu_ps(srcPtr + srcLocs[i] + 9);        /* load [R04|G04|B04|R05] - Need R04 */
+        _MM_TRANSPOSE4_PS(p128[0], p128[1], p128[2], p128[3]);  /* Contains R01-04 */
+        p128[4] = _mm_loadu_ps(srcPtr + srcLocs[i] + 12);       /* load [R05|G05|B05|R06] - Need R05 */
+        p128[5] = _mm_loadu_ps(srcPtr + srcLocs[i] + 15);       /* load [R06|G06|B06|R07] - Need R06 */
+        p128[6] = _mm_loadu_ps(srcPtr + srcLocs[i] + 18);       /* load [R07|G07|B07|R08] - Need R07 */
+        p128[7] = _mm_loadu_ps(srcPtr + srcLocs[i] + 21);       /* load [R08|G08|B08|R09] - Need R08 */
+        _MM_TRANSPOSE4_PS(p128[4], p128[5], p128[6], p128[7]);  /* Contains R05-08 */
+        p[i] = _mm256_setr_m128(p128[i], p128[ i + 4]);
+    }
+}
+
+inline void rpp_glitch_load24_i8pkd3_to_f32pln3_avx(Rpp8s *srcPtr, __m256 *p, int *srcLocs)
+{
+    __m128i px[2];
+    px[0] = _mm_loadu_si128((__m128i *)(srcPtr + srcLocs[0]));      /* load [R01|G01|B01|R02|G02|B02|R03|G03|B03|R04|G04|B04|R05|G05|B05|R06] - Need R01-04 */
+    px[1] = _mm_loadu_si128((__m128i *)(srcPtr + srcLocs[0] + 12)); /* load [R05|G05|B05|R06|G06|B06|R07|G07|B07|R08|G08|B08|R09|G09|B09|R10] - Need R05-08 */
+    p[0] = _mm256_cvtepi32_ps(_mm256_setr_m128i(_mm_shuffle_epi8(px[0], xmm_pxMaskR), _mm_shuffle_epi8(px[1], xmm_pxMaskR)));   /* Contains R01-08 */
+
+    px[0] = _mm_loadu_si128((__m128i *)(srcPtr + srcLocs[1]));      /* load [R01|G01|B01|R02|G02|B02|R03|G03|B03|R04|G04|B04|R05|G05|B05|R06] - Need G01-04 */
+    px[1] = _mm_loadu_si128((__m128i *)(srcPtr + srcLocs[1] + 12)); /* load [R05|G05|B05|R06|G06|B06|R07|G07|B07|R08|G08|B08|R09|G09|B09|R10] - Need G05-08 */
+    p[1] = _mm256_cvtepi32_ps(_mm256_setr_m128i(_mm_shuffle_epi8(px[0], xmm_pxMaskG), _mm_shuffle_epi8(px[1], xmm_pxMaskG)));   /* Contains R01-08 */
+
+    px[0] = _mm_loadu_si128((__m128i *)(srcPtr + srcLocs[2]));      /* load [R01|G01|B01|R02|G02|B02|R03|G03|B03|R04|G04|B04|R05|G05|B05|R06] - Need B01-04 */
+    px[1] = _mm_loadu_si128((__m128i *)(srcPtr + srcLocs[2] + 12)); /* load [R05|G05|B05|R06|G06|B06|R07|G07|B07|R08|G08|B08|R09|G09|B09|R10] - Need B05-08 */
+    p[2] = _mm256_cvtepi32_ps(_mm256_setr_m128i(_mm_shuffle_epi8(px[0], xmm_pxMaskB), _mm_shuffle_epi8(px[1], xmm_pxMaskB)));   /* Contains R01-08 */
+}
+
+inline void rpp_glitch_load30_u8pkd3_to_u8pkd3_avx(Rpp8u *srcPtr, int * srcLocs, __m256i &p)
+{
+    __m256i px[3];
+    px[0] = _mm256_loadu_si256((__m256i *)(srcPtr + srcLocs[0]));
```

Comment:
Please add inline comments on what is happening for these instructions

Reply:
Done